OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -1674,6 +1674,53 @@ internal static void UpgradeActorRules(ModData modData, int engineVersion, ref L
 					}
 				}
 
+				// Split aim animation logic from WithTurretAttackAnimation to separate WithTurretAimAnimation
+				if (engineVersion < 20180223)
+				{
+					var turAttackAnim = node.Value.Nodes.FirstOrDefault(n => n.Key.StartsWith("WithTurretAttackAnimation", StringComparison.Ordinal));
+					if (turAttackAnim != null)
+					{
+						var atkSequence = turAttackAnim.Value.Nodes.FirstOrDefault(n => n.Key == "AttackSequence");
+						var aimSequence = turAttackAnim.Value.Nodes.FirstOrDefault(n => n.Key == "AimSequence");
+
+						// If only AimSequence is null, just rename AttackSequence to Sequence (ReloadPrefix is very unlikely to be defined in that case).
```

Comment:
I'm confused now. Why would want to rename `AttackSequence`? You didn't change that on `WithTurretAttackAnimation`?

Reply:
> You didn't change that on WithTurretAttackAnimation?

But I wanted to, thanks for pointing this out.